op-service/sources/l1_beacon_client.go

```diff
@@ -63,6 +78,93 @@ func (cl *L1BeaconClient) apiReq(ctx context.Context, dest any, reqPath string,
 	return nil
 }
 
+func (cl *BeaconHTTPClient) NodeVersion(ctx context.Context) (string, error) {
+	var resp eth.APIVersionResponse
+	if err := cl.apiReq(ctx, &resp, versionMethod, nil); err != nil {
+		return "", err
+	}
+	return resp.Data.Version, nil
+}
+
+func (cl *BeaconHTTPClient) ConfigSpec(ctx context.Context) (eth.APIConfigResponse, error) {
+	var configResp eth.APIConfigResponse
+	if err := cl.apiReq(ctx, &configResp, specMethod, nil); err != nil {
+		return eth.APIConfigResponse{}, err
+	}
+	return configResp, nil
+}
+
+func (cl *BeaconHTTPClient) BeaconGenesis(ctx context.Context) (eth.APIGenesisResponse, error) {
+	var genesisResp eth.APIGenesisResponse
+	if err := cl.apiReq(ctx, &genesisResp, genesisMethod, nil); err != nil {
+		return eth.APIGenesisResponse{}, err
+	}
+	return genesisResp, nil
+}
+
+func (cl *BeaconHTTPClient) BeaconBlobSideCars(ctx context.Context, fetchAllSidecars bool, slot uint64, hashes []eth.IndexedBlobHash) (eth.APIGetBlobSidecarsResponse, error) {
+	reqPath := path.Join(sidecarsMethodPrefix, strconv.FormatUint(slot, 10))
+	var reqQuery url.Values
+	if !fetchAllSidecars {
+		reqQuery = url.Values{}
+		for i := range hashes {
+			reqQuery.Add("indices", strconv.FormatUint(hashes[i].Index, 10))
+		}
+	}
+	var resp eth.APIGetBlobSidecarsResponse
+	if err := cl.apiReq(ctx, &resp, reqPath, reqQuery); err != nil {
+		return eth.APIGetBlobSidecarsResponse{}, err
+	}
+	return resp, nil
+}
+
+// FallbackBeaconClient implents the BeaconClient interface. It has two underlying BeaconClients,
+// it first queries the primary and then queries the secondary if the first call fails
+type FallbackBeaconClient struct {
+	primary   BeaconClient
+	secondary BeaconClient
+}
+
+func NewFallbackClient(a, b client.HTTP) *FallbackBeaconClient {
+	return &FallbackBeaconClient{
+		primary:   &BeaconHTTPClient{cl: a},
+		secondary: &BeaconHTTPClient{cl: b},
+	}
+}
+
+func (cl *FallbackBeaconClient) NodeVersion(ctx context.Context) (string, error) {
+	if r, e := cl.primary.NodeVersion(ctx); e == nil {
+		return r, e
+	}
```

Comment:
`return r, nil` would be more readable.

Also, can we log the error, so we can tell when the primary node is failing and when we are using the secondary as fallback?

Same for the other API methods below.